tools/ci-build/changelogger/src/new.rs

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use anyhow::Context;
+use clap::Parser;
+use smithy_rs_tool_common::changelog::{FrontMatter, Markdown, Reference, Target};
+use smithy_rs_tool_common::git::find_git_repository_root;
+use smithy_rs_tool_common::here;
+use std::path::PathBuf;
+
+#[derive(Parser, Debug, Eq, PartialEq)]
+pub struct NewArgs {
+    /// Target audience for the change (if not provided, user's editor will open for authoring one)
+    #[clap(long, short = 't')]
+    pub applies_to: Option<Vec<Target>>,
+    /// List of git usernames for the authors of the change (if not provided, user's editor will open for authoring one)
+    #[clap(long, short)]
+    pub authors: Option<Vec<String>>,
+    /// List of relevant issues and PRs (if not provided, user's editor will open for authoring one)
+    #[clap(long, short)]
+    pub references: Option<Vec<Reference>>,
+    /// Whether or not the change contains a breaking change (defaults to false)
+    #[clap(long, action)]
+    pub breaking: bool,
+    /// Whether or not the change implements a new feature (defaults to false)
+    #[clap(long, action)]
+    pub new_feature: bool,
+    /// Whether or not the change fixes a bug (defaults to false)
+    #[clap(long, short, action)]
+    pub bug_fix: bool,
+    /// The changelog entry message (if not provided, user's editor will open for authoring one)
+    #[clap(long, short)]
+    pub message: Option<String>,
+    /// Basename of a changelog markdown file (defaults to a random 6-digit basename)
+    #[clap(long)]
+    pub basename: Option<PathBuf>,
+}
+
+impl From<NewArgs> for Markdown {
+    fn from(value: NewArgs) -> Self {
+        Markdown {
+            front_matter: FrontMatter {
+                applies_to: value.applies_to.unwrap_or_default().into_iter().collect(),
+                authors: value.authors.unwrap_or_default(),
+                references: value.references.unwrap_or_default(),
+                breaking: value.breaking,
+                new_feature: value.new_feature,
+                bug_fix: value.bug_fix,
+            },
+            message: value.message.unwrap_or_default(),
+        }
+    }
+}
+
+pub fn subcommand_new(args: NewArgs) -> anyhow::Result<()> {
+    let mut md_full_filename = find_git_repository_root("smithy-rs", ".").context(here!())?;
+    md_full_filename.push(".changelog");
+    md_full_filename.push(args.basename.clone().unwrap_or(PathBuf::from(format!(
+        "{}.md",
+        fastrand::u32(1_000_000..10_000_000)
```

Comment:
I actually liked the idea of using current Unix epoch time in secs as a file stem. Updated in 876da43.

Not necessarily to avoid the name collision (even with a random number, there probably wouldn't have been collisions because we have a small number of changelog entries per release, like 5), but because it gives a certain ordering among files under `.changelog` directory. If two people happened to create changelogs at the same current Unix epoch time, then it would be a collision, but it should be as less likely as using random numbers in practice.

Most importantly, the decision is reversible regardless of whether we use a random number or the current Unix epoch time for a file stem.